### ISSUE TITLE
A11y_AzureToolsStorageExplorer_ConfigureSoftDeletePolicy_ScreenReader: VoiceOver announces entire content present in screen when we tab to shift VoiceOver focus to "What is Soft Delete?" link.

### ISSUE THREAD

**Issue by vinodbk197** [microsoft/AzureStorageExplorer#1965]
“[Check out Accessibility Insights! ](https://accessibilityinsights.io/)Identify accessibility bugs before check-in and make bug fixing faster and easier.”

GitHubTags: #A11y_AzureToolsStorageExplorer; #A11yMAS; #A11YTCS; #DesktopApp; #MAS1.3.1; #ScreenReader; #VoiceOver; #Mac; 

**Environment Details:**
Application Name: Azure Tools Storage Explorer
Application Version: 1.10.0
MAC Version 10.14.6 Beta(18G29g)

**Repro Steps:**
1. Launch Storage Explorer
2. Navigate to "Open Connect dialog" pane and Sign in to Azure.
3. Go to "Manage Accounts" pane and Select the subscriptions you will use. Once selected, click 
    on "Apply" button.
4. Navigate to "Blob Containers" of the selected subscription, right click and Select "Configure
    Soft Delete Policy" and hit enter.
5. Start Voiceover.
6. Navigate to "What is Soft Delete?" link.
7. Verify Voiceover announces entire content present in screen when we tab to shift VoiceOver focus to  "What is Soft Delete?" link.

**​Actual:​**
**Issue1:** While navigating on the screen using Tab key for the first time, VoiceOver announces entire content present in screen when we tab to shift VoiceOver focus to "What is Soft Delete?" link.

- Same issue is observed for "Enable Soft Delete" radio button in this Scenario.

**Issue2:** And also for second time While navigating on the screen using Tab key, VoiceOver announces "dialog cancel button What is Soft Delete?" when VoiceOver focus is on "What is Soft Delete?".

**Same Issue1 is observed in below Scenarios:**

   **1. SAS URL**
         
Here in this Scenario VoiceOver announces entire screen content when Shared Access 
Signature screen gets opened.

**Same Issue2 is observed in below Scenarios:**

**1. Delete Popup**
      Here in this Scenario VoiceOve rannounces dialog title when shift focus from delete control 
      to cancel control.

**2. Rename blob**
      Here in this Scenario VoiceOver announces dialog title when shift focus from New name edit
      box to Rename control.

**3. Update Access Tier**
       Here in this Scenario VoiceOver announces entire screen content when Press Tab key to shift
       focus to "Learn More" link.

**4. Set Container Public Access Level**
      Here in this Scenario VoiceOver unnecessary announces other content present on the screen
      when Tab to shift keyboard focus to "No Public Access" radio button.

**5. Undelete Selected**
       Here in this Scenario VoiceOver unnecessary announces other content present on the screen
       when Tab to shift keyboard focus to "Cancel" control for the first time.

**Expected:**
**Issue1:** VoiceOver should not announce entire content present in screen when we tab to shift VoiceOver focus to "What is Soft Delete?" link, Instead VoiceOver should announce only link name with description.

**Issue2:** And also for second time While navigating on the screen using Tab key, VoiceOver should not announce" dialog cancel button What is Soft Delete?" when VoiceOver focus is on "What is Soft Delete?", Instead VoiceOver should announce only link name with description.

**User Impact:**
Screen reader users gets confusion if screen reader start announcing entire content of the screen at a time when shift screen reader focus to link.

**Recommendation:**
https://microsoft.sharepoint.com/teams/msenable/mas/Pages/browse-fixes.aspx

**MAS Reference:**
https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}

**Attachment for Reference:**
[1965_A11y_AzureToolsStorageExplorer_ConfigureSoftDeletePolicy_ScreenReader.pptx](https://github.com/microsoft/AzureStorageExplorer/files/3669444/1965_A11y_AzureToolsStorageExplorer_ConfigureSoftDeletePolicy_ScreenReader.pptx)

**Comment by JasonYeMSFT** [comment]
We have tried it on macOS 10.15.1 and it doesn't announce the entire page anymore, could you please verify that?

**Comment by JasonYeMSFT** [comment]
Related to #1908 . See the comments there.

**Comment by Suraj1200499** [comment]
@MRayermannMSFT Verified this issue in Microsoft Azure Tools Storage Explorer Version: 1.12.0, MAC Version: 10.15.3, this issue is Partially Fixed.

**Issue1**: For "What is Soft Delete Policy" link Voice over not announces entire Screen.
But when tab to shift Voice Over focus to "Enable Soft Delete" radio button, Voice Over announces content "When Soft delete is enabled, delete blobs...…….etc" along with radio button. So this issue still reproduces.
**Expected**: Voice Over Should not announce Content "When Soft delete is enabled, delete blobs...…….etc" along with radio button, it should announce only "Enable Soft Delete" selected radio button 1 of 2.

**Issue2**: Not Reproduced.

**Comment by vinodbk197** [comment]
@MRayermannMSFT Verified this issue in Private build of Mac.
Application version: 1.12.0.
This issue is no longer reproduced. Hence Closing the bug.
Added Closed Video in[ One Drive](https://microsoft-my.sharepoint.com/personal/v-nisaho_microsoft_com/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fv%2Dnisaho%5Fmicrosoft%5Fcom%2FDocuments%2FStorageExplorer%5FReprovideos&originalPath=aHR0cHM6Ly9taWNyb3NvZnQtbXkuc2hhcmVwb2ludC5jb20vOmY6L3Avdi1uaXNhaG8vRXNTVUtnZm9PSzlIbUVMOFBxcVhkOTRCVXRBVkJ6S0pldHB5SDZwS0FzVXh5dz9ydGltZT15X05kR0luRzEwZw) for reference.

**Comment by fsteffi** [comment]
GithubTags:#Closed;

**Comment by mstechie** [comment]
GitHubTags:#BM-TCS-StorageExplorer-Mac-Sep2019;
